core/src/main/java/io/aiven/klaw/helpers/db/rdbms/SelectDataJdbc.java

```diff
@@ -381,61 +389,147 @@ public List<TopicRequest> selectTopicRequestsByStatus(
       String requestor,
       String status,
       boolean showRequestsOfAllTeams,
-      int tenantId) {
-    log.debug("selectTopicRequests {} {}", requestor, status);
+      int tenantId,
+      String teamName,
+      String env,
+      String wildcardSearch) {
+    log.debug(
+        "selectTopicRequests {} {} {} {} {}", requestor, status, teamName, env, wildcardSearch);
 
     Integer teamSelected = selectUserInfo(requestor).getTeamId();
-    List<TopicRequest> topicRequests;
-
-    if (showRequestsOfAllTeams) {
-      topicRequests = topicRequestsRepo.findAllByTenantId(tenantId);
-    } else {
-      topicRequests = topicRequestsRepo.findAllByTeamIdAndTenantId(teamSelected, tenantId);
-    }
+    List<TopicRequest> topicRequests = new ArrayList<>();
+    List<TopicRequest> topicRequestListSub;
 
     if (allReqs) { // approvers
 
       List<TopicRequest> claimTopicReqs =
-          topicRequestsRepo.findAllByTopictypeAndTenantId(
-              RequestOperationType.CLAIM.value, tenantId);
-      topicRequests =
-          topicRequests.stream()
+          Lists.newArrayList(
+              findTopicRequestsByExample(
+                  RequestOperationType.CLAIM.name(),
+                  teamName,
+                  env,
+                  status,
+                  tenantId,
+                  String.valueOf(teamSelected)));
+
+      topicRequestListSub =
+          Lists.newArrayList(
+              findTopicRequestsByExample(null, teamName, env, status, tenantId, null));
+
+      // Only execute just before adding the separate claim list as this will make sure only the
+      // claim topics this team is able to approve will be returned.
+      topicRequestListSub =
+          topicRequestListSub.stream()
               .filter(
                   topicRequest ->
-                      !RequestOperationType.CLAIM.value.equals(topicRequest.getTopictype()))
-              .collect(Collectors.toList());
-      claimTopicReqs =
-          claimTopicReqs.stream()
-              .filter(
-                  claimTopicReq ->
-                      Objects.equals(claimTopicReq.getDescription(), "" + teamSelected))
+                      !RequestOperationType.CLAIM.name().equals(topicRequest.getTopictype()))
               .collect(Collectors.toList());
 
-      topicRequests.addAll(claimTopicReqs);
-
+      topicRequestListSub.addAll(claimTopicReqs);
       // remove users own requests to approve/show in the list
-      topicRequests =
-          topicRequests.stream()
+      // Placed here as it should only apply for approvers.
+      topicRequestListSub =
+          topicRequestListSub.stream()
               .filter(topicRequest -> !topicRequest.getRequestor().equals(requestor))
               .collect(Collectors.toList());
+    } else {
+      topicRequestListSub =
+          Lists.newArrayList(findTopicRequestsByExample(null, null, null, null, tenantId, null));
```

Comment:
teamId is missing here ?

Reply:
So the logic changed slightly in your last update but this was replacing just a look up by tenantId.
But from https://github.com/aiven/klaw/blob/bfc8377726a2b46e8f2c4a2f5d8dd7f88f4fbfc4/core/src/main/java/io/aiven/klaw/helpers/db/rdbms/SelectDataJdbc.java (line 428)
else {
      topicRequestListSub = Lists.newArrayList(topicRequestsRepo.findAllByTenantId(tenantId));
    }
    
    That is then filtered in Line 444
    
    ------------- 
    This is your last update
    
     if (showRequestsOfAllTeams) {
      topicRequests = topicRequestsRepo.findAllByTenantId(tenantId);
    } else {
      topicRequests = topicRequestsRepo.findAllByTeamIdAndTenantId(teamSelected, tenantId);
    }
    
I will walk through this in the IDE and see as I am actually doing more filtering in the code here then in the code from the last update.